src/frontend/src/lib/derived/token.derived.ts

```diff
@@ -29,7 +35,15 @@ export const tokenDecimals: Readable<number | undefined> = derived(
 	([$token]) => $token?.decimals
 );
 
-export const tokenToggleable: Readable<boolean> = derived(
-	[token],
-	([$token]) => nonNullish($token) && 'enabled' in $token
-);
+export const tokenToggleable: Readable<boolean> = derived([token], ([$token]) => {
+	if (nonNullish($token)) {
+		if (icTokenIcrcCustomToken($token)) {
+			return isIcrcCustomTokenEnabled($token);
+		}
+		if (icTokenEthereumUserToken($token)) {
+			return isEthereumUserTokenEnabled($token);
+		}
+	}
+
+	return false;
+});
```

Comment:
is it possible to refactor this to avoid if statements?

`nonNullissh ? icTokenIcrcCustomToken ? ... isIcrcCustomTokenEnabled `